### PR TITLE
Added mqtts documentation

### PR DESCRIPTION
**Pull request by Mjaethers**
A Guide on how to configure the MQTT channels so that they are secured

### CODE REVIEW

File: content/docs/manuals/networking-internals/mqtt-authentication.md
Comment (Malyuk-A) [inline content/docs/manuals/networking-internals/mqtt-authentication.md]:
```suggestion
and the cluster-service-manager appropriately propagates this information. Since the nodes report sensitive information, such
```

File: content/docs/manuals/networking-internals/mqtt-authentication.md
Comment (Malyuk-A) [inline content/docs/manuals/networking-internals/mqtt-authentication.md]: Nitpick:

I would use smaller headers (for the entire page - i.e. simply add one '#' more) (i.e. instead of '#', go down to '##') in the body of the markdown, because the title from line 2 (MQTT Authentication) has the same size as the other sections in this page - which IMO looks a bit strange.

E.g:
image

vs.
image

File: content/docs/manuals/networking-internals/mqtt-authentication.md
Comment (Malyuk-A) [inline content/docs/manuals/networking-internals/mqtt-authentication.md]: Suggestions:

1) It feels a bit off to start the entire page by mentioning the image that will come at the end of the paragraph. - I would place this sentence at the end of the paragraph just before showing the figure.

2) The second sentence is quite vague. What gets reported, and what information is propagated?

File: content/docs/manuals/networking-internals/mqtt-authentication.md
Comment (Malyuk-A) [inline content/docs/manuals/networking-internals/mqtt-authentication.md]: At least when I read the header "MQTT" I assume to get information about MQTT - maybe even 1-2 sentences + a link that explains what MQTT is and stands for.

Unfortunately, we do not have the main networking bits in here.
Let's image you are a new user/reader and you are not even familiar with MQTT and you stumble upon this page and read the section with the header MQTT - afterwards you still do not know what this is or what information is send between the nodes.

File: content/docs/manuals/networking-internals/mqtt-authentication.md
Comment (Malyuk-A) [inline content/docs/manuals/networking-internals/mqtt-authentication.md]:
```suggestion
MQTTS supports exchanging certificates to establish a TLS-secured channel. For this, the server (MQTT Broker) and every client require a certificate-keyfile pair signed by the same certificate Authority (CA).
The MQTT broker can be configured to only accept incoming secured connections and to identify devices by their certificate CN entry.
```

File: content/docs/manuals/networking-internals/mqtt-authentication.md
Comment (Malyuk-A) [inline content/docs/manuals/networking-internals/mqtt-authentication.md]:
```suggestion
contains many useful scripts such as ones for [creating MQTTS certificate files](https://github.com/oakestra/automation/tree/d43f701134fdf71e1206532883006e1937c38ef9/development_cluster_management/generate_mqtts_certificates). {{< /callout >}}
```

File: content/docs/manuals/networking-internals/mqtt-authentication.md
Comment (Malyuk-A) [inline content/docs/manuals/networking-internals/mqtt-authentication.md]: Suggestion:

How about following a similar approach as the other pages and using a requirements callout?
+ Making the "Getting Started" a link?
+ Emphasize on having a running Oakestra Setup instead of "assume you know how to deploy"
~
image

File: content/docs/manuals/networking-internals/mqtt-authentication.md
Comment (Malyuk-A) [inline content/docs/manuals/networking-internals/mqtt-authentication.md]:
```suggestion
Navigate into the `cluster_orchestrator` directory in the oakestra repository.
```

File: content/docs/manuals/networking-internals/mqtt-authentication.md
Comment (Malyuk-A) [inline content/docs/manuals/networking-internals/mqtt-authentication.md]:
```suggestion
1. Configure the MQTT Broker by adding the following lines to the `mosquitto/mosquitto.conf` file:
```

File: content/docs/manuals/networking-internals/mqtt-authentication.md
Comment (Malyuk-A) [inline content/docs/manuals/networking-internals/mqtt-authentication.md]:
```suggestion
        4. Export the keyfile password as an environment variable:\
```

File: content/docs/manuals/networking-internals/mqtt-authentication.md
Comment (Malyuk-A) [inline content/docs/manuals/networking-internals/mqtt-authentication.md]: https://stackoverflow.com/questions/70677153/are-they-called-environment-or-environmental-variables

```suggestion
        4. Export the keyfile password as an environment variable:\
```

File: content/docs/manuals/networking-internals/mqtt-authentication.md
Comment (Malyuk-A) [inline content/docs/manuals/networking-internals/mqtt-authentication.md]:
```suggestion
1. Copy the `ca.crt` and `ca.key` files to the worker node.
```

File: content/docs/manuals/networking-internals/mqtt.png
Comment (Malyuk-A) [inline content/docs/manuals/networking-internals/mqtt.png:1]: Nitpicks:

I am not sure if you created this figure from scratch or if you have taken it from some other place.

The transparent background bits, especially the MQTT connection between the nodes is barely visible in dark mode.

The MQTT "lollipop" connection is also distorted.

The Cluster Orchestrator component elements are not centered vertically. 

If you have access/created the original figure it would be great if you could fix those bits.

image

File: content/docs/manuals/networking-internals/mqtt-authentication.md
Comment (Malyuk-A) [inline content/docs/manuals/networking-internals/mqtt-authentication.md]:
```suggestion
       When prompted for the CN, enter the public IP of the machine
```

**Review by Malyuk-A — COMMENTED** [review]
Note regarding my comments:
Please do not feel forced to blindly follow/take-over my suggestions and nitpicks - they are subjective and optional - if you want to discuss something further go ahead 
🙂

File: content/docs/manuals/networking-internals/mqtt.png
Comment (Malyuk-A) [inline content/docs/manuals/networking-internals/mqtt.png:1]: I think this is a good first candidate to start using a more maintainable version of doing UML:

image

**Comment by Mjaethers** [comment]
Thank you very much! I made changes to reflect most of your feedback.
Regarding the more maintainable UML diagrams, I absolutely agree in principle, however as far as I can tell the suggestions in discord still require us to generate two images and then include some logic to decide which image to show based off whether we are in dark mode or not. Since mermaid is a JS tool, I don't see how we can include this in .md files (let me know if this is possible!). For now I replicated Giovanni's style from the [getstarted](https://github.com/nitindermohan/oak-new-docs/tree/getstarted) branch and included the .drawio file so that it can be adjusted fairly painlessly.
I also rebased this branch on the getstarted branch as there is a reference to those pages and the link would break as soon as that branch would be merged, so now this PR will have to wait until that branch is merged.

**Comment by Malyuk-A** [comment]
Nice changes!

We need to come up with a common understanding of how we want to do the branches and light/dark mode (especially for figures).

I will bring these topics up in the following meetings.

If I understand correctly you are building this PR up on top of the `getstarted` branch. If that is so how about making that branch the merge target instead of `main`. You can do that at the very top when editing your PR.

image

image

Otherwise, your reviewer needs to know exactly what you changed and find those changes in a large set of all `getstarted` changes - which is cumbersome.

**Comment by Malyuk-A** [comment]
Because we are already reviewing this PR I think you can safely set it to Open instead of "Draft"

File: content/docs/manuals/networking-internals/MQTT-Arch-dark.png
Comment (Malyuk-A) [inline content/docs/manuals/networking-internals/MQTT-Arch-dark.png:1]: If I am correct we cannot see this dark-mode figure in the current state of our docs.
If so how about removing this image from git?

File: content/docs/manuals/networking-internals/mqtt-authentication.md
Comment (Malyuk-A) [inline content/docs/manuals/networking-internals/mqtt-authentication.md]:
```suggestion
MQTT is a lightweight messaging protocol that supports publishing/subscribing to named channels. Oakestra uses MQTT due to its minimal network usage and low processing overhead. More information at [mqtt.org](https://mqtt.org/).
```

File: content/docs/manuals/networking-internals/mqtt-authentication.md
Comment (Malyuk-A) [inline content/docs/manuals/networking-internals/mqtt-authentication.md:28]:
```suggestion
## MQTTS
MQTT supports exchanging certificates to establish a TLS-secured channel. For this, the server (MQTT Broker) and every client require a certificate-key file pair signed against the same certificate Authority (CA). The MQTT broker can be configured to only accept incoming secured connections and to identify devices by their certificate common name (CN) entry.
```

File: content/docs/manuals/networking-internals/mqtt-authentication.md
Comment (Malyuk-A) [inline content/docs/manuals/networking-internals/mqtt-authentication.md]: For this concrete case I recommend to use `yml` instead of bash.

This is not really bash nor yaml.
But conf has no nice indentation.

bash has a strange coloring to it here

image

where as yml still provides the easy copy button and is homogeneous
image

File: content/docs/manuals/networking-internals/mqtt-authentication.md
Comment (Malyuk-A) [inline content/docs/manuals/networking-internals/mqtt-authentication.md]:
```suggestion
       ```bash
       openssl genrsa -aes256 -out client.key 2048
```

Otherwise :

image

File: content/docs/manuals/networking-internals/mqtt-authentication.md
Comment (Malyuk-A) [inline content/docs/manuals/networking-internals/mqtt-authentication.md]: The end is quite abrupt. It would be great to have 1 sentence or paragraph telling the user that e.g. with this/congrats now you have added MQTTS, or provide an example of how to verify that this is indeed working.

**Review by Malyuk-A — COMMENTED** [review]
Only a bit of polishing left IMO

File: content/docs/manuals/networking-internals/MQTT-Arch-dark.png
Comment (Mjaethers) [reply content/docs/manuals/networking-internals/MQTT-Arch-dark.png:1]: I'll see what we land on regarding dark mode images in the maintainers, maybe someone has a suggestion on how programmatically choose the correct image.

**Comment by giobart** [comment]
@Mjaethers can this be evenetually merged into main directly?

**Comment by Malyuk-A** [comment]
> @Mjaethers can this be evenetually merged into main directly?

I think this is exactly what I mentioned in the last meeting - that we need to spread the word that we no longer pursue the "intermediate" branch tactics, but we create normal feature branches and merge them into main.

Hopefully, this confusion will shortly subside. 😀

**Comment by Mjaethers** [comment]
I rebased this branch onto the getstarted branch because I reference some of those pages. We can merge this PR into main directly, but this will result in some broken links until getstarted is merged into main. To me it makes sense to just merge this into getstarted.

**Comment by Malyuk-A** [comment]
@giobart I thought we already merged the loose branches (such as getting-started) into main - but from what I was able to see this is not the case - is anything blocking this?

**Comment by giobart** [comment]
@Malyuk-A the [gettingstarted](https://github.com/nitindermohan/oak-new-docs/tree/getstarted) branch, that's where the getstarted manual for oakestra is, I'll address Nitinder's comments and open a PR about it. Unfortunately the branch name was a poor choice on my end :D

File: content/docs/manuals/networking-internals/MQTT-Arch-dark.png
Comment (Malyuk-A) [reply content/docs/manuals/networking-internals/MQTT-Arch-dark.png:1]: I think we can remove the standalone underlined MQTT text
image
(otherwise the text will not be visible in one of the modes)

Comment (Mjaethers) [reply content/docs/manuals/networking-internals/MQTT-Arch-dark.png:1]: I changed the colour to grey, which is readable in both modes

**Comment by Malyuk-A** [comment]
Thank you for realizing all the change requests! 
LGTM now

**Review by Malyuk-A — APPROVED** [review]

**Pull request merged**
